realm/realm-library/src/androidTest/java/io/realm/RealmResultsTest.java

```diff
@@ -1146,4 +1146,100 @@ public void testComparisons() {
         assertEquals(10, testRealm.where(AllTypes.class).lessThanOrEqualTo(AllTypes.FIELD_LONG, -1).findAll().size());
         assertEquals(10, testRealm.where(AllTypes.class).lessThan(AllTypes.FIELD_LONG, 0).findAll().size());
     }
+
+    private RealmResults<Dog> populateRealmResultsOnDeletedLinkView() {
+        testRealm.beginTransaction();
+        Owner owner = testRealm.createObject(Owner.class);
+        for (int i = 0; i < 10; i++) {
+            Dog dog = new Dog();
+            dog.setName("name_" + i);
+            dog.setOwner(owner);
+            owner.getDogs().add(dog);
+        }
+        testRealm.commitTransaction();
+
+
+        RealmResults<Dog> dogs = owner.getDogs().where().equalTo(Dog.FIELD_NAME, "name_0").findAll();
+        //dogs = dogs.where().findFirst().getOwner().getDogs().where().equalTo(Dog.FIELD_NAME, "name_0").findAll();
+
+        testRealm.beginTransaction();
+        owner.removeFromRealm();
+        testRealm.commitTransaction();
+        return dogs;
+    }
+
+    public void testisValidResultsBuiltOnDeletedLinkView() {
```

Comment:
We will have to remember to convert all these to JUnit4 when merging with master :(